### ISSUE TITLE
[Feature] Setting to make auto filter apply default to off

### ISSUE THREAD

**Issue by t3hmrman** [nocodb/nocodb#963]
**Need for this feature**
Putting in filter options can be a bit annoying due to the password popup appearing when an invalid filter is immediately applied, before it's been completely specified.

**Is your feature request related to a problem? Please describe.**
At present there's a bug (?) where certain inputs of filter options generate errors when executed immediately (`autoApplyFilter`) and the password prompt is displayed (I assume this is because of some error handling which assumes the error that came back was an access error? The sheet in question is not password protected).

**Describe the solution you'd like**
A way to specify the default for whether filters are auto applied. I'd like to make the default of NocoDB be to *not* apply filters immediately.

**Describe alternatives you've considered**

**Additional context**

Related code paths:

- [Frontend code governing checkbox via the `autosave` reactive variable](https://github.com/nocodb/nocodb/blob/master/packages/nc-gui/components/project/spreadsheet/components/columnFilterMenu.vue#L71)
- [Default state that seems to govern`autoApplyFilter`](https://github.com/nocodb/nocodb/blob/ba16c2d0a77bc1537e31fd301fcb01f33fd8f3f0/packages/nc-gui/store/windows.js#L39)

**Comment by o1lab** [comment]
Thank you for pointers. Is this for a shared view ? or just a regular view.

**Comment by o1lab** [comment]
We'd fixed a problem similar since you'd mentioned it in a reddit thread :)

If you can provide an example where filter options generate errors ? it would be helpful.

Also pls provide the version details of nocodb (right top menu)

**Comment by t3hmrman** [comment]
> Thank you for pointers. Is this for a shared view ? or just a regular view.

This was a shared view (those trying to view it are not logged in), but I want to note that it's the actual table as well. In trying to reproduce it I've noticed that I can break the shared view by simply inputting the filter (partially applied) -- I must remove the incomplete where clause otherwise when trying to view the data (in another private window), the password popup appears (signifying an error, if I'm guessing right)

> We'd fixed a problem similar since you'd mentioned it in a reddit thread :)

Greatly appreciate it, ya'll are so busy, and I'm getting a ton of mileage out of NocoDB [on a salary sharing site I launched](https://nomorepizzaparties.com), so I can only give my thanks!

> If you can provide an example where filter options generate errors ? it would be helpful.
> Also pls provide the version details of nocodb (right top menu)

Will do! I'm going to take a closer look, maybe do some recordings and update this issue. My version data is here, apologies should have included it right away:

```
Node: **v12.22.6**
Arch: **x64**
Platform: **linux**
Docker: **false**
Database: **pg**
ProjectOnRootDB: **false**
RootDB: **sqlite3**
PackageVersion: **0.83.8**
```

**Comment by o1lab** [comment]
This is so helpful & detailed. We will sort this in upcoming release! Thank you so much.

**Comment by t3hmrman** [comment]
No worries and no rush — the work around is easy and it’s arguably not even a bug!

As always thanks for the great software, NocoDB is awesome.

**Comment by pranavxc** [comment]
@t3hmrman: Look like you are using an older version of NocoDB(`0.83.8`) kindly please upgrade it to `0.84.6` since we had fixed it in the recent version. If the problem still persists even after the upgrade, feel free to update here.

**Comment by t3hmrman** [comment]
Hey @pranavxc thanks for the pointer -- I see you're referring to [#937](https://github.com/nocodb/nocodb/issues/937) right? That certainly fixes the problem with the password prompt showing up -- I've upgraded and I don't see that issue any more. 

I noticed that when people put in invalid filters as well it basically ignores them, so that's also much improved.

I do still want to be able to set the default for auto apply to off, but that's not a huge deal at all anymore when the UI is this much better -- most people will be able to figure out that ineffectual filters just return the whole dataset.

Thanks so much!
